lib/SPIRV/SPIRVReader.cpp

```diff
@@ -759,12 +781,15 @@ bool SPIRVToLLVM::isSPIRVCmpInstTransToLLVMInst(SPIRVInstruction *BI) const {
 // TODO: Instead of direct translation to OCL we should always produce SPIR-V
 // friendly IR and apply lowering later if needed
 bool SPIRVToLLVM::isDirectlyTranslatedToOCL(Op OpCode) const {
-  if (isSubgroupAvcINTELInstructionOpCode(OpCode) ||
-      isIntelSubgroupOpCode(OpCode))
-    return true;
+  if (isSubgroupAvcINTELInstructionOpCode(OpCode))
```

Comment:
Why do we need so many conditions and all of them do the same - `return false;`?
Should we leave just one - `if (OCLSPIRVBuiltinMap::rfind(OpCode, nullptr)) {...}` and return `false` on all other cases?

Reply:
No, because for all of them `if (OCLSPIRVBuiltinMap::rfind(OpCode, nullptr)) {...}` actually returns true. We do transition to SPIR-V friendly IR step by step so patches are small and affordable. The main idea is to fully remove `isDirectlyTranslatedToOCL` function, so everything is translated via SPIR-V friendly IR.